plugins/output/flight/flight.go

```diff
@@ -47,27 +48,86 @@ func (f *Flight) Write(metrics []telegraf.Metric) error {
 	builder := array.NewRecordBuilder(memory.DefaultAllocator, &f.schema)
 	defer builder.Release()
 
-	// Currently, the plugin only implements support for the simplest schema
-	// supported by legacy JVM and current Rust versions of [ModelarDB](https://github.com/ModelarData/ModelarDB-RS),
-	// as listed below. Support for an arbitrary schema is planned.
-
-	tidBuilder := builder.Field(0).(*array.Int32Builder) // Unused and deprecated time series identifier.
-	timeBuilder := builder.Field(1).(*array.TimestampBuilder)
-	valueBuilder := builder.Field(2).(*array.Float32Builder)
+	schemaFields := f.schema.Fields()
 
 	builder.Reserve(len(metrics))
 
 	for _, m := range metrics {
 
-		tidBuilder.AppendValues([]int32{1}, nil)
-
 		timeInt := m.Time().UnixMilli()
 
-		timeBuilder.AppendValues([]arrow.Timestamp{arrow.Timestamp(timeInt)}, nil)
-
-		fieldlst := m.FieldList()
-
-		valueBuilder.AppendValues([]float32{float32(fieldlst[0].Value.(float64))}, nil)
+		for i, f := range schemaFields {
+			switch f.Type.ID() {
+			case arrow.TIMESTAMP:
+				builder.Field(i).(*array.TimestampBuilder).AppendValues([]arrow.Timestamp{arrow.Timestamp(timeInt)}, nil)
+			case arrow.STRING:
+				tag, wasSet := m.GetTag(f.Name)
+				if !wasSet {
+					fmt.Printf("tag %d : %s not set", i, f.Name)
+				}
+				builder.Field(i).(*array.StringBuilder).AppendValues([]string{tag}, nil)
+			case arrow.INT32:
```

Comment:
it seems that the almost duplicated code in the field cases could be removed by using a helper function.

Reply:
I think this is fixed now, was this what you had in mind?

Reply:
I was more thinking if it was possible to put the entire internal switch statement into a function. The function would then take the builder and type as arguments.

Reply:
I agree with @CGodiksen that switch should probably be moved to a separate function if it cannot be generated.

Reply:
I have moved the entire loop to a different function with switches to another function with the minimal amount of parameters I could think of. Is this okay?

Reply:
I see.. Yeah, I can give you a whole writeup of why that type assertion (`(*array.Int32Builder)`) is very comlicated to implement into a seperate function without dynamic code generation. 
Instead, I think this should just be left alone for now, and postponed until a final design for this entire builder process has been made.

Reply:
Makes sense.